File: software/examples/rust/example_threshold.rs
use std::{io, error::Error};
use std::thread;
use tinkerforge::{ip_connection::IpConnection, 
                  dust_detector_bricklet::*};


const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Dust Detector Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let dd = DustDetectorBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
    // Don't use device before ipcon is connected.

		// Get threshold receivers with a debounce time of 10 seconds (10000ms).
		dd.set_debounce_period(10000);

     let dust_density_reached_receiver = dd.get_dust_density_reached_callback_receiver();

        // Spawn thread to handle received callback messages. 
        // This thread ends when the `dd` object
        // is dropped, so there is no need for manual cleanup.
        thread::spawn(move || {
            for dust_density_reached in dust_density_reached_receiver {           
                		println!("Dust Density: {} µg/m³", dust_density_reached);
            }
        });

		// Configure threshold for dust density "greater than 10 µg/m³".
		dd.set_dust_density_callback_threshold('>', 10, 0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_threshold.go
package main

import (
	"fmt"
	"tinkerforge/dust_detector_bricklet"
	"tinkerforge/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Dust Detector Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	dd, _ := dust_detector_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get threshold receivers with a debounce time of 10 seconds (10000ms).
	dd.SetDebouncePeriod(10000)

	dd.RegisterDustDensityReachedCallback(func(dustDensity uint16) {
		fmt.Printf("Dust Density: %d µg/m³\n", dustDensity)
	})

	// Configure threshold for dust density "greater than 10 µg/m³".
	dd.SetDustDensityCallbackThreshold('>', 10, 0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
